### PR TITLE
Make special attack counter respect ids and npc deaths

### LINKED ISSUE #6009: Special attack counter incorrect for Corporeal Beast and Chaos Elemental
Please check if your issue is not a duplicate by [searching existing issues](https://github.com/runelite/runelite/search?type=Issues)

**Describe the bug**
If i'm duoing with somebody else, the special attack counter cannot keep accurate count of my damage dealt.

This happens at Chaos Elemental and corp with a bgs (haven't tested other weapons).
If i hit a 40 with my spec, the counter shows ~44 damage done.

**To Reproduce**
1. Spec something with somebody else in a duo (whether they use a spec weapon or not).
2. Note your spec damage.
3. Check the amount Runelite says has been dealt.

**Expected behavior**
I expect the count to be correct

**Environment (please complete the following information):**
 - OS: OSX
 - RuneLite version: 1.5.1
 - Launcher version: (not sure how to check)

### PR DESCRIPTION
**Pull request by deathbeam**
Closes #6009

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java [hunk 1]
```diff
@@ -24,36 +24,43 @@
...
+	GENERAL_GRAARDOR(1.325d, NpcID.GENERAL_GRAARDOR, NpcID.GENERAL_GRAARDOR_6494),
+	GIANT_MOLE(1.075d, NpcID.GIANT_MOLE, NpcID.GIANT_MOLE_6499),
+	KALPHITE_QUEEN(1.05d, NpcID.KALPHITE_QUEEN, NpcID.KALPHITE_QUEEN_963, NpcID.KALPHITE_QUEEN_965, NpcID.KALPHITE_QUEEN_4303, NpcID.KALPHITE_QUEEN_4304, NpcID.KALPHITE_QUEEN_6500, NpcID.KALPHITE_QUEEN_6501),
+	KING_BLACK_DRAGON(1.075d, NpcID.KING_BLACK_DRAGON, NpcID.KING_BLACK_DRAGON_2642, NpcID.KING_BLACK_DRAGON_6502),
+	KRIL_TSUROTH(1.375d, NpcID.KRIL_TSUTSAROTH, NpcID.KRIL_TSUTSAROTH_6495),
+	VENETENATIS(1.4d, NpcID.VENENATIS, NpcID.VENENATIS_6610),
+	VETION(1.225d, NpcID.VETION, NpcID.VETION_REBORN);
 
-	private final String name;
+	private final List<Integer> ids;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: should be a Set if it's only doing a contains

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: But there is no Arrays.asSet

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: Sets.newHashSet(array)

or

new HashSet<T>(Arrays.asList(array))

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: Updated

**Pull request merged**